exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/common/HashTableTemplate.java

```diff
@@ -694,7 +694,7 @@ public PutStatus put(int incomingRowIdx, IndexPointer htIdxHolder, int hashCode,
     }
     htIdxHolder.value = currentIdx;
     return  addedBatch ? PutStatus.NEW_BATCH_ADDED :
-        (freeIndex + 1 > totalIndexSize) ?
+        (freeIndex + 1 > prevIndexSize + batchHolders.get(batchHolders.size()-1).getTargetBatchRowCount()) ?
```

Comment:
`prevIndexSize` is used in two places, here and when checking if a new batch is needed. Both use it the same way (by adding the size of the last batch). Both places are part of the **HOT** code, called for every row.
   So maybe if this variable could be set to include the size of the last batch, a lot of computations would be saved.
   Will it be simple?  Need to know the size of the last batch when setting this variable.